contributors/design-proposals/customresources-subresources.md

```diff
@@ -0,0 +1,221 @@
+
+However, it is one of [most requested features](https://github.com/kubernetes/kubernetes/issues/38113) and this proposal seeks to add  `/status `and `/scale` subresources for CustomResources.
+
+## Goals
+
+1. Support status/spec split for CustomResources:
+    1. Status changes are ignored on the main resource endpoint.
+    2. Support a `/status` subresource HTTP path for status changes.
+    3. `metadata.Generation` is increased only on spec changes.
+2. Support a `/scale` subresource for CustomResources.
+3. Maintain backward compatibility by allowing CRDs to opt-in to enable subresources.
+4. If a CustomResource is already structured using spec/status, allow it to easily transition to use the `/status` and `/scale` endpoint.
+5. Work seamlessly with [JSON Schema validation](https://github.com/kubernetes/community/pull/708).
+
+## Non-Goals
+
+1. Allow defining arbitrary subresources i.e. subresources except `/status` and `/scale`.
+2. Unify the many `Scale` types: as long as there is no generic `Scale` object in Kubernetes, we will propose to introduce yet another `Scale` type in the `apiextensions.k8s.io` api group. If Kubernetes switches to a generic `Scale` object, `apiextensions.k8s.io` will follow.
+
+## Proposed Extension of CustomResourceDefinition
+
+### API Types
+
+The addition of the following types in `apiextensions.k8s.io/v1beta1` is proposed:
+
+```go
+type CustomResourceDefinitionSpec struct {
+    Group string
+    Version string
+    Names CustomResourceDefintion
+    Scope ResourceScope
+    Validation *CustomResourceValidation
+    // SubResources describes the subresources for CustomResources
+    // This field is alpha-level and should only be sent to servers that enable
+    // subresources via the CurstomResourceSubResources feature gate.
+    // +optional
+    SubResources *CustomResourceSubResources `json:“subResources,omitempty”`
+}
+
+// CustomResourceSubResources defines the status and scale subresources for CustomResources.
+type CustomResourceSubResources {
+    // Status denotes the status subresource for CustomResources
+    Status *CustomResourceSubResourceStatus `json:“status,omitempty”`
+    // Scale denotes the scale subresource for CustomResources
+    Scale  *CustomResourceSubResourceScale  `json:“scale,omitempty”`
+}
+
+// CustomResourceSubResourceStatus defines how to serve the HTTP path <CR Name>/status.
+type CustomResourceSubResourceStatus struct {
+    // The JSON path (e.g. “.status”) of the status of a CustomResource.
+    // The whole object is transferred over the wire, but only the status can be mutated via the /status subresource.
+    // StatusPath is restricted to be “.status” and defaults to “.status”.
+    StatusPath string `json:“statusPath,omitempty”`
+    // The JSON path (e.g. “.spec”) of the spec of a CustomResource.
+    // SpecPath is restricted to be “.spec” and defaults to “.spec”.
```

Comment:
> We want the possibility to add fields later (i.e. no bool, but a struct) and we want to avoid an empty struct. Any better idea?

If you make it a pointer (like it is), doesn't the presence or absence of the field trigger you?